catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/provider/SolrProviderQuery.java

```diff
@@ -0,0 +1,1649 @@
+
+    assertEquals(0, response.getResults().size());
+
+    // NUMERICAL FORMAT
+    Filter numericalFilter =
+        filterBuilder.attribute("missingField").greaterThanOrEqualTo().number(23L);
+
+    response = provider.query(quickQuery(numericalFilter));
+
+    assertEquals(0, response.getResults().size());
+  }
+
+  @Test
+  public void testQueryMissingSortField() throws IngestException, UnsupportedQueryException {
+
+    deleteAll();
+
+    MockMetacard m = new MockMetacard(Library.getTampaRecord());
+    m.setTitle("Tampa");
+
+    List<Metacard> list = Arrays.asList(m, new MockMetacard(Library.getFlagstaffRecord()));
+
+    create(list);
+
+    Filter txtFilter = filterBuilder.attribute("id").like().text("*");
+
+    QueryImpl query = new QueryImpl(txtFilter);
+
+    query.setSortBy(new ddf.catalog.filter.impl.SortByImpl("unknownField", SortOrder.ASCENDING));
+
+    SourceResponse response = provider.query(new QueryRequestImpl(query));
+
+    assertEquals(2, response.getResults().size());
+  }
+
+  /**
+   * Testing if the temporal search does not fail when no schema field can be found and/or there is
+   * no data in the index
+   */
+  @Test
+  public void testQueryMissingSortFieldTemporal()
+      throws IngestException, UnsupportedQueryException {
+
+    /*
+     * I have tested this with an empty schema and without an empty schema - both pass, but
+     * there is no regression test for the empty schema scenario TODO there should probably be
+     * an automated test that creates a fresh cache, that would be a better test
+     */
+
+    deleteAll();
+
+    Filter txtFilter = filterBuilder.attribute("id").like().text("*");
+
+    QueryImpl query = new QueryImpl(txtFilter);
+
+    query.setSortBy(new ddf.catalog.filter.impl.SortByImpl(Result.TEMPORAL, SortOrder.ASCENDING));
+
+    SourceResponse response = provider.query(new QueryRequestImpl(query));
+
+    assertEquals(0, response.getResults().size());
+  }
+
+  /**
+   * If parts of a query can be understood, the query should be executed whereas the part that has a
+   * missing property should return 0 results.
+   */
+  @Test
+  public void testTwoQueriesWithOneMissingField() throws Exception {
+
+    deleteAll();
+
+    MockMetacard m = new MockMetacard(Library.getTampaRecord());
+    m.setTitle("Tampa");
+
+    List<Metacard> list = Arrays.asList(m, new MockMetacard(Library.getFlagstaffRecord()));
+
+    create(list);
+
+    Filter filter =
+        filterBuilder.anyOf(
+            filterBuilder.attribute(Metacard.TITLE).text("Tampa"),
+            filterBuilder.attribute("missingField").text("someText"));
+
+    SourceResponse sourceResponse = provider.query(new QueryRequestImpl(new QueryImpl(filter)));
+
+    assertEquals("Tampa should be found", 1, sourceResponse.getResults().size());
+  }
+
+  @Test(expected = IngestException.class)
+  public void testDeleteNullAttribute() throws IngestException {
+
+    provider.delete(
+        new DeleteRequest() {
```

Comment:
❓ Should we extract and reuse those instead of creating new anonymous inner classes every time?